apps/functions-subscription/src/adapters/azure/cosmosdb/subscription-history.ts

```diff
@@ -0,0 +1,23 @@
+import { pipe } from 'fp-ts/lib/function';
+import * as E from 'fp-ts/lib/Either';
+import * as TE from 'fp-ts/lib/TaskEither';
+import { Database } from '@azure/cosmos';
+import { SubscriptionHistoryWriter } from '../../../domain/subscription-history';
+import { cosmosErrorToDomainError } from './errors';
+
+export const makeSubscriptionHistoryCosmosContainer = (
+  db: Database,
+): SubscriptionHistoryWriter => {
+  const container = db.container('subscription-history');
+  return {
+    insert: (subscriptionHistory) =>
+      pipe(
+        TE.tryCatch(
+          () => container.items.create(subscriptionHistory),
```

Comment:
why not extends [versioned_model](https://github.com/pagopa/io-functions-commons/blob/master/src/utils/cosmosdb_model_versioned.ts) with already embedded versioning functions ? see :[https://github.com/pagopa/io-functions-commons/blob/bc42d61eb4124769d2385de0dcd3ddc907f6e6a1/src/models/service.ts#L265](https://github.com/pagopa/io-functions-commons/blob/bc42d61eb4124769d2385de0dcd3ddc907f6e6a1/src/models/service.ts#L265)

Reply:
Good question! For now, it seemed simpler to implement just the insert function without additional layers and remain consistent with how [subscription adapter is implemented](https://github.com/pagopa/trial-system/blob/92d9358/apps/functions-subscription/src/adapters/azure/cosmosdb/subscription.ts). We can consider migrating the adapters related to Cosmos to `CosmosdbModel*`, I would do that as a separate task.